sky/execution.py

```diff
@@ -234,8 +234,8 @@ def _execute(
     # Requested features that some clouds support and others don't.
     requested_features = set()
 
-    if task.num_nodes > 1:
-        requested_features.add(clouds.CloudImplementationFeatures.MULTI_NODE)
+    # Add requested features from the task
+    requested_features |= task.get_requested_features()
```

Comment:
How about renaming it to `get_required_cloud_features` as what we did in https://github.com/skypilot-org/skypilot/blob/99d0affd1b43f437a704e1aa9280c3ce355cb93c/sky/resources.py#L1154-L1169